packages/portal-subgraph/src/portal.ts

```diff
@@ -0,0 +1,136 @@
+import { BigInt, log } from "@graphprotocol/graph-ts"
+import {
+  Portal,
+  EscrowSettled,
+  EscrowSlashed,
+  OrderCancelled,
+  OrderMatched,
+  OrderPlaced,
+  OwnerUpdated,
+  ParamUpdated
+} from "../generated/Portal/Portal"
+import { Escrow, Order } from "../generated/schema"
+import { ESCROW_STATUS_PENDING, ESCROW_STATUS_SETTLED, ESCROW_STATUS_SLASHED, ORDER_STATUS_PENDING, ORDER_STATUS_FILLED, ORDER_STATUS_CANCELLED } from "./utils/constants";
+
+export function handleEscrowSettled(event: EscrowSettled): void {
+  let escrowID = event.params.escrowID.toString();
+  let escrow = Escrow.load(escrowID);
+
+  // Entities only exist after they have been saved to the store;
+  // `null` checks allow to create entities on demand
```

Comment:
We're not creating on demand here--maybe something like

```
// Escrow should always exist by the time EscrowSettled emits
```

Reply:
Yea part of the starter code I need to clean up